### PR TITLE
Boostrap marconi-tutorial package

### PR DESCRIPTION
**Pull request by koslambrou**
* Provide a simple AddressCount indexer with the new Marconi API
* Tutorial is an executable which starts the indexing with an HTTP server (which doesn't actually return the query result of the indexer for now)
* Renamed functions in Marconi.ChainIndex.CLI with more significant names
* Refactored the marconi-chain-index experimental main application

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [ ] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [ ] Targeting main unless this is a cherry-pick backport
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [ ] Reviewer requested

### CODE REVIEW

**Review by berewt — APPROVED** [review]
LGTM, comments contain more answers about the existing design than code improvements.

File: marconi-tutorial/src/Marconi/Tutorial/Indexers/AddressCount.hs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+  deriving (Show)
+
+type instance Point AddressCountEvent = C.ChainPoint
+
+newtype AddressCountQuery = AddressCountQuery C.AddressAny
+  deriving (Eq, Ord, Show)
+
+type instance Result AddressCountQuery = Int
+
+-- Not idea that the 'Maybe' type is forced here?
```
[the remark this answers is not in the document]
Comment (koslambrou) [reply marconi-tutorial/src/Marconi/Tutorial/Indexers/AddressCount.hs:78]: Let me rephrase the advantage. So if I provide `Nothing`, then this block event will _not_ be used in the batching. However, if I use `Just []`, then it's just inefficient because it adds an "empty" element in the batch?

[the remark this answers is not in the document]
Comment (berewt) [reply marconi-tutorial/src/Marconi/Tutorial/Indexers/AddressCount.hs:78]: Yes, when we had just `event`, without maybe, we passed everything to the `inDatabase` indexer, leading to a SQL transaction even if no rows can be inserted. Also, in the general case, an event may be something else than a list.
